### PR TITLE
chore: api call env set-up

### PR DESCRIPTION
**Pull request by aziyatali**
Tried to set initial api call, there are lots of things I do not know. Just sharing progress. To be continued tomorrow!
Also, need to restructure them so we can get function lists with single or a few exports!

### CODE REVIEW

**Comment by aziyatali** [comment]
@Jscripter-pk , would you please review the code structure, inheritance class and give some feedback please?
Available endpoints in olive has been implemented so far. 
After, I will ask @JonghunYu regarding structure whether we are moving all the endpoint calls here so we can use them for new-loan and other mobile apps.

**Review by Jscripter-pk — COMMENTED** [review]
@aziyatali I think QueryType is going to be dynamic. As We will be passing whatever will be required to pass.
And As I can see you have added jest config. I would suggest basic test for the functions you wrote. Atleast the test that give some confidence that what you are doing is working

**Comment by aziyatali** [comment]
@JonghunYu님, would you please review the src folder where functions for all the endpoints were implemented. Also, handled the case client case. 
Test cases to be implemented(in progress)

File: package.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
     "test": "jest",
     "build": "rm -rf dist && webpack",
     "lint": "eslint .",
-    "format:write": "prettier --write ."
+    "prettier": "prettier --write -c .prettierrc.json"
```
Comment (JonghunYu) [inline package.json:14]: rename this `format` as it cause conflict with running prettier itself

Comment (JonghunYu) [reply package.json:14]: and I think you don't need to specify config file with `-c .prettierrc.json` as it's a default path

File: __test__/auth.test.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  }
+
+  describe('/auth/login (POST)', () => {
+    it('returns 201 with valid HMAC Authorization header', () => {
+      const accessId = '2ZAK7ZDWMDQ1LD32'
+      const secretKey = 'h3tfeJI6K9iAXloanvZ6'
+      const date = new Date().toUTCString()
+      const hashString = hmacHashString(secretKey, [date, 'post'].join(' '))
+
+      return request(app.getHttpServer())
```
Comment (JonghunYu) [inline __test__/auth.test.ts:23]: this won't work

File: package.json [hunk 3]
```diff
@@ -58,8 +58,12 @@
   },
   "homepage": "https://github.com/ninetydays/sign-SDK#readme",
   "dependencies": {
+    "@faker-js/faker": "^7.3.0",
+    "@nestjs/common": "^9.2.1",
```
Comment (JonghunYu) [inline package.json:62]: do you have any reason to install this?

File: src/api/loans.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
+import { KrossClient } from '../kross-client'
+import { QueryType } from '../types'
+
+export class Accounts extends KrossClient {
```
Comment (JonghunYu) [inline src/api/loans.tsx:4]: Loans?

File: src/kross-client.ts [hunk 5]
```diff
@@ -1,35 +1,88 @@
...
+              ...config.headers,
+              'client-authorization': hmacToken,
+              'X-Date': xDate,
+              authorization: `Bearer ${this.authToken}`,
+            })
+          : (config.headers = {
+              ...config.headers,
+              'client-authorization': hmacToken,
+              'X-Date': xDate,
+            })
```
Comment (JonghunYu) [inline src/kross-client.ts:27]:
```ts
config.headers = {
  ...config.headers,
  'client-authorization': hmacToken,
  'X-Date': xDate,
};

if (this.authToken) {
  config.headers = {
    ...config.headers,
    authorization: `Bearer ${this.authToken}`,
  };
}
```

File: src/types/index.ts [hunk 6]
```diff
@@ -8,3 +10,35 @@ export interface WindowPopUpParams {
...
+  limit?: any
+  sortBy?: string
+  bankId?: string
+  accountNumber?: number
+  loanId?: number
+  investmentId?: number
+  amount?: number
+  idempotencyKey?: number
+  loan_id?: number
+}
```
Comment (JonghunYu) [inline src/types/index.ts:34]: You shouldn't add specific fields only used in some endpoints but define them as a separate type

**Review by JonghunYu — CHANGES_REQUESTED** [review]
@aziyatali 

@Jscripter-pk Please review this PR too

File: package.json (hunk 3 above)
Comment (aziyatali) [reply package.json:62]: I was planning to use them for mock testing purpose for invalid data of predefined data(to generate invalid credentials).

File: __test__/auth.test.ts (hunk 2 above)
Comment (aziyatali) [reply __test__/auth.test.ts:23]: Yes, it was the test case in the beginning which to be updated soon

File: src/api/loans.tsx (hunk 4 above)
Comment (aziyatali) [reply src/api/loans.tsx:4]: Gotcha!

File: src/kross-client.ts [hunk 7]
```diff
@@ -1,21 +1,16 @@
 import axios, { AxiosInstance, AxiosRequestConfig } from 'axios'
 import { createHmac } from 'crypto'
-
-export type KrossClientOptions = AxiosRequestConfig & {
-  accessId: string
-  secretKey: string
-}
+import { queryType } from './types'
+import { KrossClientOptions } from './types/index'
```
[the remark this answers is not in the document]
Comment (aziyatali) [reply src/kross-client.ts]: currently outtdated

File: src/kross-client.ts [hunk 8]
```diff
@@ -1,35 +1,88 @@
...
+              ...config.headers,
+              'client-authorization': hmacToken,
+              'X-Date': xDate,
+            })
         return config
       },
       (error) => Promise.reject(error)
     )
+
+    this.client.interceptors.response.use(
```
Comment (Jscripter-pk) [inline src/kross-client.ts:33]: @aziyatali What do you think of creating a function with a proper name that defines the purpose of this piece of code

File: src/api/investments.tsx [hunk 9]
```diff
@@ -0,0 +1,76 @@
+import { KrossClient } from '../kross-client'
+import { QueryType } from '../types'
+
+export class Investments extends KrossClient {
+  notes = async (params: QueryType) => {
+    this.method = 'get'
```
[the remark this answers is not in the document]
Comment (aziyatali) [reply src/api/investments.tsx:6]: It used method for hashing the hmac. What do you suggest?

File: src/kross-client.ts (hunk 8 above)
Comment (aziyatali) [reply src/kross-client.ts:37]: Islam, I think leaving them as they are better maybe, it handles intercepts response

**Comment by aziyatali** [comment]
@JonghunYu @Jscripter-pk , Would please review it? 
Currently implemented mock testing just for login and users.

**Comment by Jscripter-pk** [comment]
@aziyatali mocking is not required here use mock responses in case of kftc / third party services

**Review by JonghunYu — APPROVED** [review]

**Pull request merged**